src/structures/vroom/input/input.cpp

```diff
@@ -645,6 +649,7 @@ void Input::set_matrices(unsigned nb_thread) {
   for (const auto& profile : _profiles) {
     thread_profiles[t_rank % nb_buckets].push_back(profile);
     ++t_rank;
+    #if USE_ROUTING
```

Comment:
A plain removal of this portion is a bit extreme. As a result, we hit the assert on line 667 when running on a "usual" problem with no custom matrix:

```
vroom -i docs/example_1.json
vroom: structures/vroom/input/input.cpp:677: vroom::Input::set_matrices(unsigned int)::<lambda(const std::vector<std::__cxx11::basic_string<char> >&)>: Assertion `d_m != _durations_matrices.end()' failed.
Aborted (core dumped)
```

The expected behavior here would be to get a nice and explicit error. Maybe we could keep `add_routing_wrapper` after all but simply have it throw with "VROOM compiled without routing support" a bit like here:

https://github.com/VROOM-Project/vroom/blob/e3938092e60e1b2d2c5da58feef0b5fe819cd5f4/src/structures/vroom/input/input.cpp#L88